tagger.py

```diff
@@ -0,0 +1,55 @@
+import os
+import pydicom
+import argparse
+
+
+class TagError(Exception):
+    pass
+
+
+def add_send_tag(dicomfile, group, identifier, tag_value):
+    for elem_tag in range(0x0010, 0x00FF):
+        dataelem = dicomfile.get((group, elem_tag))
+        if dataelem:
+            is_private_creator = dataelem.tag.group == group and 0x0010 <= dataelem.tag.element <= 0x00ff
+            if is_private_creator and dataelem.value.lower() == identifier.lower():
+                for private_tag_element in range(dataelem.tag.elem * 0x0100, dataelem.tag.elem * 0x0100 + 0x0100):
+                    private_tag = (dataelem.tag.group, private_tag_element)
+                    private_elem = dicomfile.get(private_tag)
+                    if not private_elem:
+                        dicomfile.add_new(private_tag, 'LO', tag_value)
+                        return
+                    elif private_elem.value.lower().startswith(tag_value.lower()):
+                        return
+        else:
+            identifier_tag = (group, elem_tag)
+            dicomfile.add_new(identifier_tag, 'LO', identifier)
+            private_tag = (group, elem_tag * 0x0100)
+            dicomfile.add_new(private_tag, 'LO', tag_value)
+            return
+    raise TagError('No free element in group {} to tag the dicom'.format(group))
```

Comment:
Without calling `dicomfile.save_as(file_path)` the changes (if any) aren't committed to the file on disk.